helix-core/src/main/java/org/apache/helix/manager/zk/ParticipantManager.java

```diff
@@ -101,6 +101,10 @@ public ParticipantManager(HelixManager manager, HelixZkClient zkclient, int sess
    * @throws Exception
    */
   public void handleNewSession() throws Exception {
```

Comment:
Anyone external customer uses this class? This class is supposed to be internal class, can you confirm it? We can directly remove the method if no external customer refers this class.

Reply:
@lei-xia I did a quick check code search on LinkedIn's internal codebase. It doesn't seem to be used within LinkedIn. But not sure about external users outside linkedin. I am hesitated to just remove it since it is public. That is why we spent so much effort on backward compatibility. If we all agree on removing this deprecated method, I will remove it. @jiajunwang @kaisun2000 @narendly

Reply:
I re-think about this. I don't think we need to change this method. I don't want to create a new method handleNewSession(sessionId). But would pass sessionId to the participant manager when creating the participant manager. Because each time ZkHelixManager.handleNewSessionAsParticipant() would create a new participant manager. I think it is safe to just pass sessionId to participant manager. So I could keep minimal changes, given the correctness.

Reply:
we can not make sure we won't change methods from any public classes, that will mean we won't be able to change anything. We should treat this class as not a API, so it should be fine to change the method.